src/codecs/farbfeld.rs

```diff
@@ -259,12 +259,12 @@ impl<W: Write> FarbfeldEncoder<W> {
 
     /// Encodes the image ```data``` (native endian)
     /// that has dimensions ```width``` and ```height```
-    pub fn encode(self, data: &[u8], width: u32, height: u32) -> ImageResult<()> {
-        self.encode_impl(data, width, height)?;
+    pub fn encode<F: FnMut(Progress)>(self, data: &[u8], width: u32, height: u32, progress: F) -> ImageResult<()> {
```

Comment:
Isn't this an API breaking change?

Reply:
Sure seems like it. We could rewrite this part such that only new methods are added and none of the existing are modified